File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-9.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        unordered_map<char,int> set;
        int maxF = 0;
        int res = -1;
        int l = 0;
        for(int r = 0; r < s.size();r++){
            set[s[r]]++;
            maxF = max(maxF,set[s[r]]);
            while((r-l+1)-maxF > k){
                set[s[l]]--;l++;
            }
            res = max(res,r-l+1);
        }
        return res;
    }
};

File: Data Structures & Algorithms/trapping-rain-water/submission-9.py
class Solution:
    def trap(self, height: List[int]) -> int:
        if not height:
            return 0
        l,r = 0 , len(height)-1
        maxLeft,maxRight = height[l],height[r]
        res = 0
        while l < r:
            if height[l] < height[r]:
                l+=1
                maxLeft = max(height[l],maxLeft)
                res += maxLeft - height[l]
            else:
                r-=1
                maxRight = max(maxRight,height[r])
                res += maxRight - height[r]
        return res